packages/flutter_tools/lib/src/cache.dart

```diff
@@ -21,16 +19,16 @@ import 'base/logger.dart';
 import 'base/net.dart';
 import 'base/os.dart' show OperatingSystemUtils;
 import 'base/platform.dart';
-import 'base/process.dart';
 import 'base/user_messages.dart';
 import 'build_info.dart';
 import 'convert.dart';
-import 'dart/package_map.dart';
-import 'dart/pub.dart';
 import 'features.dart';
```

Comment:
`features.dart` is the last unmigrated import.  The `DevelopmentArtifact.feature` property only seems to be used to check if it's enabled in `precache`.
https://github.com/flutter/flutter/blob/ce318b7b539e228b806f81b3fa7b33793c2a2685/packages/flutter_tools/lib/src/commands/precache.dart#L158-L160

We could move the static `DevelopmentArtifact` out to another file.

In `cache.dart`:
```dart
class DevelopmentArtifact {
  const DevelopmentArtifact(this.name, {this.isEnabled});
  final String name;
  final bool isEnabled;
```

In `flutter_cache.dart`:
```dart
class DevelopmentArtifacts {
  DevelopmentArtifacts(FeatureFlags featureFlags)
      : androidGenSnapshot = DevelopmentArtifact('android_gen_snapshot', feature: featureFlags.isEnabled(flutterAndroidFeature)),
        androidMaven = DevelopmentArtifact('android_maven', feature: featureFlags.isEnabled(flutterAndroidFeature)),
...
  ;

  /// Artifacts required for Android development.
  final DevelopmentArtifact androidGenSnapshot;
  final DevelopmentArtifact androidMaven;
...
```

Reply:
Not a bad idea

Reply:
or, we could split up the non-migraty parts of features.dart

Reply:
Oh actually the only thing that's keeping features from being migrated is `version.dart` and that's only used in `FlutterFeatureFlags`, which can be pulled out of `features.dart` its own file exactly like this PR does.  That's less churn, I'll do that now.

Reply:
(unless you already started on that)

Reply:
🐢
🐢
🐢
🐢
🐢
🐢

Reply:
https://github.com/flutter/flutter/pull/79603 👍